From 422ea7227a86a65cba6591a061f7f9d2bcdf4021 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Tue, 10 Oct 2023 14:40:22 +0200
Subject: [PATCH 0053/1695] drivers: gic: use DT bindings

Use DT bindings GIC_PPI and GIC_SIP instead of 1 and 0 raw values.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 287fec2de95d11bfcdbcf65a2ecc70a541c28360 drivers: gic: use DT bindings]
Change-Id: I6aa53d3452047aa8365af4f96ec7cdec18c02d61
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
[etienne: cherry-picked from 8c7282be2bf86536ba1658f7a2040ec06304fdfb]
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 core/drivers/gic.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/core/drivers/gic.c b/core/drivers/gic.c
index 555d94f97..04b4fc466 100644
--- a/core/drivers/gic.c
+++ b/core/drivers/gic.c
@@ -6,6 +6,7 @@
 
 #include <arm.h>
 #include <assert.h>
+#include <dt-bindings/interrupt-controller/arm-gic.h>
 #include <config.h>
 #include <compiler.h>
 #include <drivers/gic.h>
@@ -208,10 +209,10 @@ static int gic_dt_get_irq(const uint32_t *properties, int count, uint32_t *type,
 	it_num = fdt32_to_cpu(properties[1]);
 
 	switch (fdt32_to_cpu(properties[0])) {
-	case 1:
+	case GIC_PPI:
 		it_num += 16;
 		break;
-	case 0:
+	case GIC_SPI:
 		it_num += 32;
 		break;
 	default:
-- 
2.47.2

